libs/src/services/discoveryProvider/index.js

```diff
@@ -544,19 +544,29 @@ class DiscoveryProvider {
 
     try {
       const response = await axios(axiosRequest)
-      const {
-        data,
-        latest_indexed_block: indexedBlock,
-        latest_chain_block: chainBlock
-      } = Utils.parseDataFromResponse(response)
-
-      if ((chainBlock - indexedBlock) > UNHEALTHY_BLOCK_DIFF) {
-        const endpoint = await this.autoSelectEndpoint()
-        this.setEndpoint(endpoint)
-        throw new Error(`Selected endpoint was too far behind. Indexed: ${indexedBlock} Chain: ${chainBlock}`)
+      const parsedResponse = Utils.parseDataFromResponse(response)
+
+      if (
+        'latest_indexed_block' in parsedResponse &&
+        'latest_chain_block' in parsedResponse
+      ) {
+        const {
+          latest_indexed_block: indexedBlock,
+          latest_chain_block: chainBlock
+        } = parsedResponse
+
+        if (
+          !chainBlock ||
+          !indexedBlock ||
+          (chainBlock - indexedBlock) > UNHEALTHY_BLOCK_DIFF
+        ) {
+          const endpoint = await this.autoSelectEndpoint()
+          this.setEndpoint(endpoint)
```

Comment:
one thought - do we want to consider moving this re-selection logic into the catch after all 3 retries have been exhausted with a single dp? likeif something falls behind 20blocks briefly that's prob ok since it should recover (although shorrt term that's not the case). also since all creatornodes are hardcoded to a single discprov, if that one falls behind this would immediately cause the dependent creatornode to attempt to autoselect and then fail to select any and just throw. it would ideally just wait for some longer threshold to let it catch up (or maybe not). i think it's fine to still merge this bc we can just tune the block threshold from 20 to modify tolerance as needed, and with this change we should really modify the cnode whitelists to all discprovs as soon as we've confirmed this feature works dappside.

Reply:
yes

Reply:
@raymondjacobson will defer to whatever you want to do on this